packages/core/src/external/aws/lambda-logic/bundle-to-html.ts

```diff
@@ -1470,16 +1463,24 @@ function createVitalsByDate(observations: Observation[]): string {
     const observationDate = formatDateForDisplay(observation.effectiveDateTime);
     const existingObservation = acc.find(observation => observation.date === observationDate);
 
+    if (!observationDate.length) return acc;
+
     if (existingObservation) {
       existingObservation.observations.push(observation);
       return acc;
     }
 
-    acc.push({
-      date: observationDate,
-      observations: [observation],
+    const observationDisplay = observation.code?.coding?.find(coding => {
+      return coding.display;
     });
 
+    if (observationDisplay || observation.code?.text) {
+      acc.push({
+        date: observationDate,
+        observations: [observation],
+      });
+    }
```

Comment:
The reasoning behind this is because its a filter to prevent empty row without display being rendered

Reply:
I see. The problem with this approach is that we're bringing the display logic to the reducer, not to mention a check here doesn't mean it matches what's being displayed. It would be better to have a type that requires that the output of the reducer has the data needed for display.